### DISCUSSION TITLE
Crash.  assertion failed: std::empty(outbuf)

### DISCUSSION THREAD

**Discussion by dantearmok** [transmission/transmission#5418] [Support]
Mac mini (late 2012, 2.5-GHz 4-Core i5), 16GB RAM, plus an assortment of USB & FW HDs.
macOS 10.15 Catalina, ClamXAV, PeakHour, PIA, and Xcode 12.3 (all fully updated).

Running Transmission-7d86d67bc7 Build #9019 (Apr 16, 2023, 8:35:19 PM)

Process:               Transmission [9991]
Path:                  /Applications/Transmission.app/Contents/MacOS/Transmission
Identifier:            org.m0k.transmission
Version:               4.1.0-dev (14714.4.1.0)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Transmission [9991]
User ID:               502

Date/Time:             2023-04-18 02:47:35.830 -0400
OS Version:            Mac OS X 10.15.7 (19H2026)
Report Version:        12
Anonymous UUID:        9D83084F-22F2-20FB-B0D2-056E94905F2F

Time Awake Since Boot: 16000 seconds

System Integrity Protection: enabled

Crashed Thread:        1

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'assertion failed: std::empty(outbuf) (/Users/mac-pbbllgyava/workspace/trunk-mac/libtransmission/handshake.cc:628)'
terminating with uncaught exception of type NSException
abort() called

Application Specific Backtrace 1:
0   CoreFoundation                      0x00007fff32057a17 __exceptionPreprocess + 250
1   libobjc.A.dylib                     0x00007fff6abbca9e objc_exception_throw + 48
2   CoreFoundation                      0x00007fff32057875 +[NSException raise:format:] + 181
3   Transmission                        0x000000010711cd1e _Z16tr_assert_reportNSt3__117basic_string_viewIcNS_11char_traitsIcEEEElS3_ + 280
4   Transmission                        0x00000001070e4052 _ZN12tr_handshake7read_iaEP9tr_peerIo + 2576
5   Transmission                        0x00000001070e48f1 _ZN12tr_handshake8can_readEP9tr_peerIoPvPm + 679
6   Transmission                        0x00000001070e7a1e _ZN9tr_peerIo16can_read_wrapperEv + 164
7   Transmission                        0x00000001070e7cc4 _ZN9tr_peerIo8try_readEm + 292
8   Transmission                        0x00000001070e6d80 _ZN9tr_peerIo13event_read_cbEisPv + 216
9   Transmission                        0x000000010717fbe0 event_process_active_single_queue + 502
10  Transmission                        0x000000010717cceb event_base_loop + 1134
11  Transmission                        0x00000001070dc035 _ZN22tr_session_thread_impl17sessionThreadFuncEP10event_base + 87
12  Transmission                        0x00000001070dc427 _ZNSt3__1L14__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEM22tr_session_thread_implFvP10event_baseEPS7_S9_EEEEEPvSE_ + 63
13  libsystem_pthread.dylib             0x00007fff6bf65109 _pthread_start + 148
14  libsystem_pthread.dylib             0x00007fff6bf60b8b thread_start + 15

Thread 0:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff6be9ddea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6be9e160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff31fdbd07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff31fda7c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff31fd9c33 CFRunLoopRunSpecific + 466
5   com.apple.HIToolbox           	0x00007fff30bf5aad RunCurrentEventLoopInMode + 292
6   com.apple.HIToolbox           	0x00007fff30bf57c5 ReceiveNextEventCommon + 584
7   com.apple.HIToolbox           	0x00007fff30bf5569 _BlockUntilNextEventMatchingListInModeWithFilter + 64
8   com.apple.AppKit              	0x00007fff2f23e3c9 _DPSNextEvent + 883
9   com.apple.AppKit              	0x00007fff2f23cc10 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
10  com.apple.AppKit              	0x00007fff2f22e91e -[NSApplication run] + 658
11  com.apple.AppKit              	0x00007fff2f200706 NSApplicationMain + 777
12  libdyld.dylib                 	0x00007fff6bd5ccc9 start + 1

Thread 1 Crashed:
0   libsystem_kernel.dylib        	0x00007fff6bea432a __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff6bf64e60 pthread_kill + 430
2   libsystem_c.dylib             	0x00007fff6be2b808 abort + 120
3   libc++abi.dylib               	0x00007fff69090458 abort_message + 231
4   libc++abi.dylib               	0x00007fff690818bf demangling_terminate_handler() + 262
5   libobjc.A.dylib               	0x00007fff6abbea57 _objc_terminate() + 96
6   libc++abi.dylib               	0x00007fff6908f887 std::__terminate(void (*)()) + 8
7   libc++abi.dylib               	0x00007fff690921a2 __cxxabiv1::failed_throw(__cxxabiv1::__cxa_exception*) + 27
8   libc++abi.dylib               	0x00007fff69092169 __cxa_throw + 113
9   libobjc.A.dylib               	0x00007fff6abbcbcc objc_exception_throw + 350
10  com.apple.CoreFoundation      	0x00007fff32057875 +[NSException raise:format:] + 181
11  org.m0k.transmission          	0x000000010711cd1e tr_assert_report(std::__1::basic_string_view<char, std::__1::char_traits<char> >, long, std::__1::basic_string_view<char, std::__1::char_traits<char> >) + 280
12  org.m0k.transmission          	0x00000001070e4052 tr_handshake::read_ia(tr_peerIo*) + 2576
13  org.m0k.transmission          	0x00000001070e48f1 tr_handshake::can_read(tr_peerIo*, void*, unsigned long*) + 679
14  org.m0k.transmission          	0x00000001070e7a1e tr_peerIo::can_read_wrapper() + 164
15  org.m0k.transmission          	0x00000001070e7cc4 tr_peerIo::try_read(unsigned long) + 292
16  org.m0k.transmission          	0x00000001070e6d80 tr_peerIo::event_read_cb(int, short, void*) + 216
17  org.m0k.transmission          	0x000000010717fbe0 event_process_active_single_queue + 502
18  org.m0k.transmission          	0x000000010717cceb event_base_loop + 1134
19  org.m0k.transmission          	0x00000001070dc035 tr_session_thread_impl::sessionThreadFunc(event_base*) + 87
20  org.m0k.transmission          	0x00000001070dc427 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >(void*) + 63
21  libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
22  libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff6bea0746 __semwait_signal + 10
1   libsystem_c.dylib             	0x00007fff6be23eea nanosleep + 196
2   libc++.1.dylib                	0x00007fff6906c6da std::__1::this_thread::sleep_for(std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > const&) + 73
3   org.m0k.transmission          	0x0000000107118a40 tr_web::Impl::curlThreadFunc() + 448
4   org.m0k.transmission          	0x0000000107118cb4 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_web::Impl::*)(), tr_web::Impl*> >(void*) + 59
5   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff6bea2756 kevent + 10
1   org.m0k.transmission          	0x000000010708ae7c -[VDKQueue watcherThread:] + 203
2   com.apple.Foundation          	0x00007fff34695222 __NSThread__start__ + 1064
3   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
4   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 4:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff6be9ddea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6be9e160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff31fdbd07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff31fda7c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff31fd9c33 CFRunLoopRunSpecific + 466
5   com.apple.AppKit              	0x00007fff2f3dfbf4 _NSEventThread + 132
6   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
7   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 5:: com.apple.coreaudio.AQClient
0   libsystem_kernel.dylib        	0x00007fff6be9ddea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6be9e160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff31fdbd07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff31fda7c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff31fd9c33 CFRunLoopRunSpecific + 466
5   com.apple.audio.toolbox.AudioToolbox	0x00007fff30350f60 GenericRunLoopThread::Entry(void*) + 166
6   libAudioToolboxUtility.dylib  	0x00007fff686e4611 CAPThread::Entry(CAPThread*) + 77
7   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 6:
0   libsystem_kernel.dylib        	0x00007fff6be9de26 semaphore_wait_trap + 10
1   com.apple.audio.caulk         	0x00007fff658a0b16 caulk::mach::semaphore::wait() + 16
2   com.apple.audio.caulk         	0x00007fff658a09b2 caulk::semaphore::timed_wait(double) + 106
3   com.apple.audio.caulk         	0x00007fff658a07c4 caulk::concurrent::details::worker_thread::run() + 30
4   com.apple.audio.caulk         	0x00007fff658a01e4 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
5   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 7:: AMCP Logging Spool
0   libsystem_kernel.dylib        	0x00007fff6be9de26 semaphore_wait_trap + 10
1   com.apple.audio.caulk         	0x00007fff658a0b16 caulk::mach::semaphore::wait() + 16
2   com.apple.audio.caulk         	0x00007fff658a09b2 caulk::semaphore::timed_wait(double) + 106
3   com.apple.audio.caulk         	0x00007fff658a07c4 caulk::concurrent::details::worker_thread::run() + 30
4   com.apple.audio.caulk         	0x00007fff658a01e4 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
5   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 8:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 9:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 10:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 11:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 12:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 1 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x000070000237b000  rcx: 0x0000700002378ac8  rdx: 0x0000000000000000
  rdi: 0x0000000000005707  rsi: 0x0000000000000006  rbp: 0x0000700002378af0  rsp: 0x0000700002378ac8
   r8: 0x0000700002378990   r9: 0x0000700002378b60  r10: 0x000070000237b000  r11: 0x0000000000000246
  r12: 0x0000000000005707  r13: 0x0000003000000008  r14: 0x0000000000000006  r15: 0x0000000000000016
  rip: 0x00007fff6bea432a  rfl: 0x0000000000000246  cr2: 0x00000001073845e5
  
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133

**Comment by ckerr** [comment]
Converted to issue @ https://github.com/transmission/transmission/issues/5423. Locking discussion thread.
